src/2025h1/const-trait.md

```diff
@@ -0,0 +1,101 @@
+
+### The status quo
+
+People write a lot of code that will be run in compile time. They include procedural macros, build scripts ([42.8k hits][build scripts] on GitHub for `build.rs`), and const functions/consts ([108k hits][const fns] on GitHub for `const fn`). Not being able to write const functions with generic behavior is often cited as a pain point of Rust's compile time capabilities. Because of the limited expressiveness of `const fn`, people may decide to move some compile time logic to a build script, which could increase build times, or simply choose not to do it in compile time (even though it would have helped runtime performance).
+
+There are also language features that require the use of traits, such as iterating with `for` and handling errors with `?`. Because the `Iterator` and `Try` traits currently cannot be used in constant contexts, people are unable to use `?` to handle results, nor use iterators e.g. `for x in 0..5`.
+
+[build scripts]: https://github.com/search?q=path%3A**%2Fbuild.rs+NOT+is%3Afork&type=code
+[const fns]: https://github.com/search?q=%22const+fn%22+language%3Arust+NOT+is%3Afork&type=code
+
+### The next 6 months
+
+The primary goal is to do "Everything but" stabilization of this feature over the next 6 months
+
+* Author an RFC for `const traits` and get it accepted by the language design team
+* Implement the syntax and user experience described in the RFC
+* Issue a call for experimentation and resolve any issues found
+* Author a stabilization report
+
+As a secondary goal, we will use this as an experiment to drive forward a-mir-formality, with @nikomatsakis and @oli-obk mentoring @tiif to extend a-mir-formality with support for const traits. 
+
+### The "shiny future" we are working towards
+
+We're working towards enabling developers to do more things in general within a `const` context. Const traits is a blocker for many future possibilities (see also the const eval [feature skill tree]) including heap operations in const contexts. 
+
+[feature skill tree]: https://rust-lang.github.io/const-eval/skill_tree.html
+
+## Design axioms
+
+None.
+
+## Ownership and team asks
+
+Steps towards the primary goal of doing everything towards stabilization apart from approving stabilization itself:
+
+| Task                               | Owner(s) or team(s)  | Notes                                                           |
+|------------------------------------|----------------------|-----------------------------------------------------------------|
+| Lang-team experiment               | ![Team][] [lang]     | ![Complete][]                                                   |
+| Experimental implementation        | @compiler-errors     | ![Complete][]                                                   |
```

Comment:
and fee1-dead, who has contributed significantly to the existing impl

Reply:
And done all the previous impl experiments which gave us lots of insights into what is possible and reasonable

Reply:
Thanks for the corrections. Will update!

Reply:
Fixed.